oci/private/tarball.sh.tpl

```diff
@@ -1,38 +1,48 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit -o nounset
 
-readonly STAGING_DIR=$(mktemp -d)
 readonly YQ="{{yq}}"
+readonly TAR="{{tar}}"
 readonly IMAGE_DIR="{{image_dir}}"
-readonly BLOBS_DIR="${STAGING_DIR}/blobs"
 readonly TARBALL_PATH="{{tarball_path}}"
 readonly TAGS_FILE="{{tags}}"
 
+# Write tar manifest in mtree format
+# https://man.freebsd.org/cgi/man.cgi?mtree(8)
+# so that tar produces a deterministic output.
+mtree=$(mktemp)
+function add_to_tar() {
+    content=$1
+    tar_path=$2
+    local mtree_flags="uid=0 gid=0 mode=0755 time=1672560000 type=file"
```

Comment:
one nit: we should create parent directories recursively, for once, for any file to match the behavior of `tar -cvf DIR OUTPUT`.

Reply:
okay, maybe that's a helper starlark function that a bazel-lib 2.x release could include, then we can reuse that logic? I assume we'll need to create parent directories there too.

Reply:
we created this functionality there but only for automatically generated mtrees. we need to this in bash here however, which can't be shared with bazel-lib as we don't use bash there to generate the mtree